File: C/2/10.c
#include <stdio.h>

/**
 * A function that converts an ASCII character to lower case.
 * 
 * @param  int c The character to be converted
 * 
 * @return int   Resultant character
 */
int lower(int c)
{
    return (c >= 'A' && c <= 'Z') ? c + 'a' - 'A' : c;
}

/**
 * A program to test bitcount.
 */
main()
{
    printf("lower('a')\t=>\t%c\n", lower('a'));
    printf("lower('b')\t=>\t%c\n", lower('b'));
    printf("lower('c')\t=>\t%c\n", lower('c'));
    printf("lower('d')\t=>\t%c\n", lower('d'));
    printf("lower('e')\t=>\t%c\n", lower('e'));
    printf("lower('A')\t=>\t%c\n", lower('A'));
    printf("lower('B')\t=>\t%c\n", lower('B'));
    printf("lower('C')\t=>\t%c\n", lower('C'));
    printf("lower('D')\t=>\t%c\n", lower('D'));
    printf("lower('E')\t=>\t%c\n", lower('E'));
    /** 
     * Following example only throws compiler warnings.
     * 'printf("lower(❤️)\t=>\t%c\n", lower("❤️"));'
     */
}

File: rustlang/2/10.rs
/// A function that converts an ASCII character to lower case.
/// In Rust, we could just use 'to_ascii_lowercase', but we are not lazy.
/// 
/// @param  char c The character to be converted
/// 
/// @return char   Resultant character
fn lower(c : char) -> char
{
    // We have to cast char to u8 to get its ASCII value...
    // ...and cast back to char to return.
    let digit_value = c as u8;

    // Rust doesn't have ternary operator, but the if-else syntax is
    // pretty flexible
    return if
                digit_value >= ('A' as u8) 
                && digit_value <= ('Z' as u8) 
            { 
                (digit_value + ('a' as u8) - ('A' as u8)) as char
            } else {
                digit_value as char
            };
}

/// A program to test bitcount.
fn main()
{
    print!("lower('a')\t=>\t{}\n", lower('a'));
    print!("lower('b')\t=>\t{}\n", lower('b'));
    print!("lower('c')\t=>\t{}\n", lower('c'));
    print!("lower('d')\t=>\t{}\n", lower('d'));
    print!("lower('e')\t=>\t{}\n", lower('e'));
    print!("lower('A')\t=>\t{}\n", lower('A'));
    print!("lower('B')\t=>\t{}\n", lower('B'));
    print!("lower('C')\t=>\t{}\n", lower('C'));
    print!("lower('D')\t=>\t{}\n", lower('D'));
    print!("lower('E')\t=>\t{}\n", lower('E'));

    // Compilation errors
    // print!("lower(❤️)\t=>\t{}\n", lower("❤️"));
}